api/src/State/PersonalInvitationRejectProcessor.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use App\DTO\PersonalInvitation;
+use App\Entity\CampCollaboration;
+use App\Entity\User;
+use App\Repository\CampCollaborationRepository;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\NonUniqueResultException;
+use Doctrine\ORM\NoResultException;
+use Symfony\Bundle\SecurityBundle\Security;
+
+/**
+ * @implements ProcessorInterface<PersonalInvitation,PersonalInvitation>
+ */
+class PersonalInvitationRejectProcessor implements ProcessorInterface {
+    public function __construct(
+        private readonly CampCollaborationRepository $campCollaborationRepository,
+        private readonly UserRepository $userRepository,
+        private readonly Security $security,
+        private readonly EntityManagerInterface $em,
+    ) {}
+
+    /**
+     * @param PersonalInvitation $data
+     *
+     * @throws NoResultException
+     * @throws NonUniqueResultException
+     */
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): PersonalInvitation {
+        $user = $this->getUser();
+
+        $campCollaboration = $this->campCollaborationRepository->findByUserAndIdAndInvited($user, $data->id);
+        $campCollaboration->status = CampCollaboration::STATUS_INACTIVE;
```

Comment:
I would throw a validation exception here if $campCollaboration is null

Reply:
[2fbe07b](https://github.com/ecamp/ecamp3/pull/4865/commits/2fbe07b4ae9a907eed17f60b7c009e026308e89f)
I also wrote a test for the only way this could happen: If you reject an existing personal invitation which does not belong to you.